colab/nessie-iceberg-flink-demo-nba.ipynb

```diff
@@ -0,0 +1,559 @@
+    "!pip install -i https://test.pypi.org/simple/ nessiedemo\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Setup the Demo: installs the required Python dependencies, downloads the sample datasets and\n",
+    "# downloads + starts the Nessie-Quarkus-Runner.\n",
+    "from nessiedemo.demo import setup_demo\n",
+    "demo = setup_demo(\"nessie-0.6-iceberg-flink-0.11.yml\", [\"nba\"])\n",
+    "\n",
+    "# This is separate, because NessieDemo.prepare() via .start() implicitly installs the required dependencies.\n",
+    "# Downloads Flink, and sets up a default table_env that points to `main`\n",
+    "from nessiedemo.flink import flink_for_demo\n",
+    "table_env, demo_flink = flink_for_demo(demo)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Nessie branches (via Nessie CLI)\n",
+    "----------------------------\n",
+    "Once all dependencies are configured, we can get started with `Nessie` by using the\n",
+    "Nessie CLI, which allows interaction with Nessie branches and tables. We'll get started with the following steps:\n",
+    "\n",
+    "- Create a new branch named `dev`\n",
+    "- List all branches\n",
+    "\n",
+    "It is worth mentioning that we don't have to explicitly create a `main` branch, since it's the default branch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a new `dev` branch\n",
+    "!nessie branch dev\n",
+    "\n",
+    "# Table environment for `dev` branch, which will be used later on\n",
+    "dev_table_env = demo_flink.table_env_for_ref(\"dev\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# List all branches with their revisions. We should see the `main` & `dev` branches.\n",
+    "!nessie --verbose branch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create tables under dev branch\n",
+    "-------------------------------------\n",
+    "Once we created the `dev` branch and verified that it exists, we can use the `dev_table_env` Table environment (that points to the `dev` branch)\n",
+    "to create some tables and add some data.\n",
+    "\n",
+    "We create two tables under the `dev` branch using the `dev_table_env`:\n",
+    "- `salaries`\n",
+    "- `totals_stats`\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the dataset\n",
+    "dataset = demo.fetch_dataset(\"nba\")\n",
+    "\n",
+    "from pyflink.table import *\n",
+    "from pyflink.table.descriptors import *\n",
+    "\n",
+    "# Creating `salaries` table\n",
+    "dev_table_env.connect(FileSystem().path(dataset['salaries.csv']))\\\n",
```

Comment:
Why do we need to create a temporary table first? (maybe worth a comment)

Reply:
because that seems to be the only workable approach in Flink. I haven't gotten it to work with any other approach

Reply:
I didn't think it was worth documenting it as people familiar with Flink would know what it means. Also as mentioned, there's no other way to load CSV data into a table

Reply:
Then maybe add explicit imports, so people see where these classes come from.

Reply:
done